sa/storage-authority.go

```diff
@@ -484,6 +548,33 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 		return
 	}
 
+	var challs []challModel
+	_, err = tx.Select(
+		&challs,
+		"SELECT * FROM challenges WHERE authorizationID = :authID ORDER BY id ASC",
+		map[string]interface{}{"authID": authz.ID},
+	)
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+	if len(challs) != len(authz.Challenges) {
```

Comment:
Right, won't it be easier to do #638 if we just make them a func now? Or are there small changes between them I missed?

Would show up as less churn in the diffs